src/client.rs

```diff
@@ -0,0 +1,850 @@
+use jsonrpc::error::Error;
+
+use super::primitives;
+
+pub struct Client {
+	agent: jsonrpc::client::Client
+}
+
+impl Client {
+	pub fn new(host: &str) -> Client {
+		Client { agent: jsonrpc::client::Client::new(host.to_owned(), None, None) }
+	}
+
+	pub fn new_with_credentials(host: &str, username: &str, password: &str) -> Client {
+		Client { agent: jsonrpc::client::Client::new(host.to_owned(), Some(username.to_owned()), Some(password.to_owned())) }
+	}
+
+	/// Returns a list of addresses owned by client.
+	///
+	/// # Arguments
+	///
+	/// * `none`
+	/// 
+	/// # Returns
+	/// 
+	/// Vector of accounts owned by the client.
+	///
+	/// # Example
+	///
+	/// ```
+	/// use nimiq_rpc::Client;
+	/// let client = Client::new("http://seed-host.com:8648");
+	/// let result = client.accounts();
+	/// ```
+	pub fn accounts(&self) -> Result<Vec<primitives::Account>, Error>{
+		let request = self.agent.build_request("accounts", &[]);
+    match self.agent.send_request(&request).and_then(|res| res.into_result::<Vec<primitives::Account>>()) {
+			Ok(r) => Ok(r),
+			Err(e) => Err(e)
+    }
+	}
+
+	/// Returns the height of most recent block.
+	///
+	/// # Arguments
+	///
+	/// * `none`
+	/// 
+	/// # Returns
+	/// 
+	/// The current block height the client is on.
+	///
+	/// # Example
+	///
+	/// ```
+	/// use nimiq_rpc::Client;
+	/// let client = Client::new("http://seed-host.com:8648");
+	/// let result = client.block_number();
+	/// ```
+	pub fn block_number(&self) -> Result<u64, Error>{
+		let request = self.agent.build_request("blockNumber", &[]);
+    match self.agent.send_request(&request).and_then(|res| res.into_result::<u64>()) {
```

Comment:
Same here, simply write `self.agent.send_request(&request).and_then(|res| res.into_result::<u64>())` instead of the match.

Reply:
Also applies to the other functions.

Reply:
Actually, you can probably simplify the whole by writing:
```
pub fn block_number(&self) -> Result<u64, Error> {
    self.agent.do_rpc("blockNumber", &[])
}
```